aws-lc-rs/src/fips.rs

```diff
@@ -0,0 +1,147 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+/// Retrieve the FIPS module service status.
+#[must_use]
+pub fn get_fips_service_status() -> FipsServiceStatus<()> {
+    #[cfg(feature = "fips")]
+    {
+        if indicator::is_approved() {
+            FipsServiceStatus::ApprovedMode(())
+        } else {
+            FipsServiceStatus::NonApprovedMode(())
+        }
+    }
+    #[cfg(not(feature = "fips"))]
+    FipsServiceStatus::NotEnabled(())
+}
+
+/// Clears the FIPS module service status. Useful if a thread has executed an unapproved
+/// algorithm, but you desire to reset the module status without destroying the thread.
+#[cfg(feature = "fips")]
+pub fn reset_fips_service_status() {
+    indicator::reset();
+}
+
+#[cfg(feature = "fips")]
+pub(crate) mod indicator {
+    use std::cell::Cell;
+
+    thread_local! {
+        static APPROVED_OPERATING_MODE: Cell<bool> = const { Cell::new(true)};
```

Comment:
does this mean that at initialisation time, the service indicator is set to "approved"? Can't figure out the Rust :)